types/modules/producer.d.ts

```diff
@@ -0,0 +1,87 @@
+import { Connection } from './connection';
+import { ProducerHooks } from './hooks';
+import type amqp = require('amqplib');
+import pDefer = require('p-defer');
+
+declare class ProducerError extends Error {
+  constructor(error: { name: string; message: string });
+}
+
+interface PublishOptions extends amqp.Options.Publish {
+  routingKey?: string;
+  rpc?: boolean;
+}
+
+declare class Producer {
+  constructor(connection: Connection);
+  hooks: ProducerHooks;
```

Comment:
```suggestion
  private hooks: ProducerHooks;
```

Reply:
These are actually public so you can add hooks to the producer, and it's exposed on the arnavmq object.